external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSource.scala

```diff
@@ -441,13 +468,12 @@ private[kafka010] case class KafkaSource(
               try {
                 result = Some(body)
               } catch {
-                case x: OffsetOutOfRangeException =>
```

Comment:
`OffsetOutOfRangeException` only happens when we call `seek` than `poll`. So it won't happen here. It's pretty confusing as it doesn't update attempt, so I just removed it.